service/src/main/java/bio/terra/pipelines/dependencies/stairway/StairwayJobService.java

```diff
@@ -216,12 +220,23 @@ public Stairway getStairway() {
   }
 
   @Traced
-  public FlightState retrieveJob(UUID jobId) {
+  public FlightState retrieveJob(UUID jobId, String userId) {
     try {
-      return stairwayComponent.get().getFlightState(jobId.toString());
+      FlightState result = stairwayComponent.get().getFlightState(jobId.toString());
+      // Note: after implementing TSPS-134, we can filter by flightId in enumerateJobs and remove
+      // the following check
+      if (!userId.equals(
```

Comment:
so i wonder if this scenario shoudl return a job not found or a "forbidden" exception.  Also at some point we will be doing these checks at the workspace level as opposed to the user level.  Is that something we want to do now or do it in another round of refactors

Reply:
re: checks at workspace level, I'd say make that change in the future if we do it, since that's only applicable if users are submitting data from a workspace, versus uploading the data in the no-BP model. or am i missing something?